lib/src/main/java/com/auth0/jwt/algorithms/RSAAlgorithm.java

```diff
@@ -55,8 +62,65 @@
         this(new CryptoHelper(), id, algorithm, keyProvider);
     }
 
+    @Override
+    public void verifyWithX509(DecodedJWT jwt, EncodeType encodeType, String jwksFile, String pemFile) throws Exception {
+        List<byte[]> byteArrayList = decode(jwt, encodeType);
```

Comment:
Why is it that we have to run "decode" again on a "decoded jwt"? I think there could be better naming here.

Reply:
fetchContentAndSignatureByteArrays()?

Reply:
Sounds good